src/controllers/keyboard/keyboardeventfilter.cpp

```diff
@@ -1,19 +1,37 @@
 #include "controllers/keyboard/keyboardeventfilter.h"
 
+#include <QAction>
 #include <QEvent>
 #include <QKeyEvent>
 #include <QtDebug>
 
 #include "moc_keyboardeventfilter.cpp"
 #include "util/cmdlineargs.h"
+#include "widget/wbasewidget.h"
 
-KeyboardEventFilter::KeyboardEventFilter(ConfigObject<ConfigValueKbd>* pKbdConfigObject,
+KeyboardEventFilter::KeyboardEventFilter(UserSettingsPointer pConfig,
+        QLocale& locale,
         QObject* parent,
         const char* name)
         : QObject(parent),
-          m_pKbdConfigObject(nullptr) {
+          m_pConfig(pConfig),
+          m_locale(locale),
+          m_enabled(false) {
     setObjectName(name);
-    setKeyboardConfig(pKbdConfigObject);
+
+    // get enabled state
+    if (pConfig->getValueString(ConfigKey("[Keyboard]", "Enabled")).length() == 0) {
+        pConfig->set(ConfigKey("[Keyboard]", "Enabled"), ConfigValue(1));
+    }
+    m_enabled = pConfig->getValue<bool>(ConfigKey("[Keyboard]", "Enabled"));
```

Comment:
Do we not want this to be enabled by default? I understand it used to be always on previously (`src/coreservices.cpp:459`).
```suggestion
    m_enabled = pConfig->getValue(ConfigKey("[Keyboard]", "Enabled"), true);
```